### PR TITLE
[MOBILE-1091] make react module attributes-compatible on ios and android

### PR DESCRIPTION
**Pull request by crow**
### What do these changes do?
Add the ability to set attributes from react-native apps using the react-native-module.

### Why are these changes necessary?
To add attributes to iOS and Android via the react-native platfrom

### How did you verify these changes?
Manual tested on iOS and Android

#### Verification Screenshots:
Manual testing Android:
https://slack-files.com/T025Q1VP7-FR6N38N75-b79d460d92

Manual testing iOS:
image

### CODE REVIEW

File: js/AttributeEditor.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+   * Adds string attribute.
+   * @instance
+   * @memberof AttributeEditor
+   * @function setString
+   *
+   * @param {string} value The attribute value.
+   * @param {string} name The attribute name.
+   * @return {AttributeEditor} The attribute editor instance.
+   */
+  setString(value: string, name: String): AttributeEditor {
```
Comment (rlepinski) [inline js/AttributeEditor.js]: I think we can get away with this taking a String, Boolean, or Number in the future under a single `set` call

File: android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java [hunk 2]
```diff
@@ -74,11 +73,16 @@
...
     private static final String TAG_OPERATION_TYPE = "operationType";
     private static final String TAG_OPERATION_TAGS = "tags";
-
     private static final String TAG_OPERATION_ADD = "add";
     private static final String TAG_OPERATION_REMOVE = "remove";
     private static final String TAG_OPERATION_SET = "set";
 
+    private static final String ATTRIBUTE_OPERATION_KEY = "key";
+    private static final String ATTRIBUTE_OPERATION_VALUE = "value";
+    private static final String ATTRIBUTE_OPERATION_ACTION = "action";
```
Comment (marc-scig) [inline android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java]: any chance we could call this ATTRIBUTE_OPERATION_TYPE/"type" so as not to further overload "action"?

File: android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java [hunk 3]
```diff
@@ -838,6 +856,37 @@ private static void applyTagGroupOperations(@NonNull TagGroupsEditor editor, @No
...
+            String value = operation.getString(ATTRIBUTE_OPERATION_VALUE);
+
+            if (action == null || key == null || value == null) {
+                continue;
+            }
+
+            if (ATTRIBUTE_OPERATION_SET.equals(action)) {
+                editor.setAttribute(key, value);
+            } else if (ATTRIBUTE_OPERATION_REMOVE.equals(action)) {
+                editor.setAttribute(key, value);
```
Comment (marc-scig) [inline android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java]: should this be remove?

Comment (crow) [reply android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java]: why yes

File: android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java (hunk 2 above)
Comment (crow) [reply android/src/main/java/com/urbanairship/reactnative/UrbanAirshipReactModule.java]: changed

File: js/AttributeEditor.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  /**
+   * Removes the attribute.
+   * @instance
+   * @memberof AttributeEditor
+   * @function removeAttribute
+   *
+   * @param {string} name The name of the attribute to remove.
+   * @return {AttributeEditor} The attribute editor instance.
+   */
+  removeAttribute(name: string): AttributeEditor {
```
Comment (rlepinski) [inline js/AttributeEditor.js]: should this be just `remove` or should we make the set `setAttribute`?

Comment (crow) [reply js/AttributeEditor.js]: changed set to setAttribute

**Pull request merged**